### PR TITLE
fix(datePipe): short timezone not displayed, closes #9812

### LINKED ISSUE #9812: DatePipe short timezone not displayed
**I'm submitting a ...**  (check one with "x")
```
[x] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

When providing format `hh:mm z` it not display timezone.

**Expected/desired behavior**

When providing format `hh:mm z` it will display timezone.

**Reproduction of the problem**
If the current behavior is a bug or you can illustrate your feature request better with an example, please provide the steps to reproduce and if possible a minimal demo of the problem via 
see _http://plnkr.co/edit/ILqRxtg9pV7A7i6Nn2Py?p=preview_

**What is the expected behavior?**

see _http://plnkr.co/edit/ILqRxtg9pV7A7i6Nn2Py?p=preview_

**What is the motivation / use case for changing the behavior?**
Fill gap between a1 dateFilter and current datePipe

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.3
 
* **Browser:** [all] 
 
* **Language:** [all]

### PR DESCRIPTION
**Pull request by laskoviymishka**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

See #9812

**What is the new behavior?**

See _http://plnkr.co/edit/st43EpvVTCSUHOFALsBk?p=preview_

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

Closes #9812

### CODE REVIEW

File: modules/@angular/facade/src/intl.ts [hunk 1]
```diff
@@ -139,6 +139,16 @@ function hourExtracter(inner: (date: Date, locale: string) => string): (
   };
 }
 
+function timeZoneGetter(timezone: string): (date: Date, locale: string) => string {
+  return function(date: Date, locale: string): string {
+    // To workaraund `Intl` API restriction for single timezone let format with 24 hours
+    let format = {hour: '2-digit', hour12: false, timeZoneName: timezone};
+    let result = new Intl.DateTimeFormat(locale, format).format(date);
```
Comment (vicb) [inline modules/@angular/facade/src/intl.ts]: use const over let. format could be moved out of the closure

Comment (laskoviymishka) [inline modules/@angular/facade/src/intl.ts]: fixed

**Comment by laskoviymishka** [comment]
@vicb done please review

File: modules/@angular/facade/src/intl.ts [hunk 2]
```diff
@@ -139,6 +139,16 @@ function hourExtracter(inner: (date: Date, locale: string) => string): (
   };
 }
 
+function timeZoneGetter(timezone: string): (date: Date, locale: string) => string {
+  const format = {hour: '2-digit', hour12: false, timeZoneName: timezone};
+  return function(date: Date, locale: string): string {
+    // To workaraund `Intl` API restriction for single timezone let format with 24 hours
```
Comment (vicb) [inline modules/@angular/facade/src/intl.ts]: workaround (typo)

**Comment by laskoviymishka** [comment]
@vicb fix type, please review

**Comment by vicb** [comment]
Thanks

**Pull request merged**